src/server/scripts/Shadowlands/Sanctum of Domination/boss_sylvanas_windrunner.cpp

```diff
@@ -0,0 +1,3702 @@
+            ridingCopy->SetUnitFlags(UNIT_FLAG_IMMUNE_TO_NPC);
+            ridingCopy->SetUnitFlags(UNIT_FLAG_NOT_SELECTABLE);
+        }
+    }
+
+    void EnterEvadeMode(EvadeReason /*why*/) override
+    {
+        Talk(SAY_DISENGAGE);
+
+        _EnterEvadeMode();
+        summons.DespawnAll();
+        _instance->SendEncounterUnit(ENCOUNTER_FRAME_DISENGAGE, me);
+        _instance->SetBossState(DATA_SYLVANAS_WINDRUNNER, FAIL);
+
+        instance->DoRemoveAurasDueToSpellOnPlayers(SPELL_BARBED_SHOT);
+        instance->DoRemoveAurasDueToSpellOnPlayers(SPELL_BANSHEE_MARK);
+        instance->DoRemoveAurasDueToSpellOnPlayers(SPELL_DOMINATION_CHAIN_PLAYER);
+        instance->DoRemoveAurasDueToSpellOnPlayers(SPELL_DOMINATION_CHAIN_PERIODIC);
+        instance->DoRemoveAurasDueToSpellOnPlayers(SPELL_VEIL_OF_DARKNESS_ABSORB_AURA);
+
+        if (Creature* bolvar = _instance->GetCreature(DATA_BOLVAR_FORDRAGON_PINNACLE))
+            bolvar->DespawnOrUnsummon();
+
+        if (Creature* jaina = _instance->GetCreature(DATA_JAINA_PROUDMOORE_PINNACLE))
+            jaina->DespawnOrUnsummon();
+
+        if (Creature* thrall = _instance->GetCreature(DATA_THRALL_PINNACLE))
+            thrall->DespawnOrUnsummon();
+
+        _DespawnAtEvade();
+    }
+
+    void Reset() override
+    {
+        _Reset();
+    }
+
+    void DamageTaken(Unit* /*attacker*/, uint32& damage) override
+    {
+        if (me->HealthBelowPctDamaged(83.0f, damage))
+        {
+            if (events.IsInPhase(PHASE_ONE) && !events.IsInPhase(PHASE_INTERMISSION))
+            {
+                events.Reset();
+
+                events.SetPhase(PHASE_INTERMISSION);
+            }
+        }
+    }
+
+    void JustSummoned(Creature* summon) override
+    {
+        summons.Summon(summon);
+
+        switch (summon->GetEntry())
+        {
+            case NPC_BOLVAR_FORDRAGON_PINNACLE:
+            {
+                summon->GetGUID();
+                break;
+            }
+
+            case NPC_SYLVANAS_SHADOW_COPY_FIGHTERS:
+            {
+                _shadowCopyGUID.push_back(summon->GetGUID());
+
+                summon->SetReactState(REACT_PASSIVE);
+
+                summon->SetUnitFlags(UNIT_FLAG_IMMUNE_TO_PC);
+                summon->SetUnitFlags(UNIT_FLAG_IMMUNE_TO_NPC);
+                summon->SetUnitFlags(UNIT_FLAG_NOT_SELECTABLE);
+                break;
+            }
+
+            case NPC_DOMINATION_ARROW:
+            {
+                summon->GetGUID();
+
+                summon->SetReactState(REACT_PASSIVE);
+
+                summon->SetUnitFlags(UNIT_FLAG_IMMUNE_TO_PC);
+                summon->SetUnitFlags(UNIT_FLAG_IMMUNE_TO_NPC);
+                summon->SetUnitFlags(UNIT_FLAG_NOT_SELECTABLE);
+                summon->SetUnitFlags2(UNIT_FLAG2_DISABLE_TURN);
+            }
+
+            default:
+                break;
+        }
+    }
+
+    void KilledUnit(Unit* victim) override
+    {
+        if (!victim && !victim->IsPlayer())
```

Comment:
Potential crash there, you need use is

if (!victim || !victim->IsPlayer()).

The && will trigger !victim->IsPlayer() in cases where the victim is null

Reply:
Does it really need a null check?

Reply:
Apparently, null is not possible on this void.